### ISSUE TITLE
Fix S3878 FP: Jagged arrays

### ISSUE THREAD

**Issue by ThorstenReichert** [SonarSource/sonar-dotnet#8163]
### Description

When a C# method has an array of arrays as `params` parameter, S3878 is raised when the method is called. e.g. for

```csharp
public class Repro8163
{
    public static void Method(params object[][] array) { }

    public static void CallParamsMethod()
    {
        Method(new[] { new object() }); // Noncompliant - FP
    }
}
```
Following the S3878 suggestion to "remove the array creation and simply pass the elements" will result in a compiler error (type mismatch "cannot convert from object to object[]").

### Expected behavior

No issue is raised.

### Actual behavior

S3878 is raised.

### Known workarounds

Move the array creation to a separate variable, then pass that variable as method parameter.

### Related information

* Visual Studio version: 17.6.1
* MSBuild / dotnet version: .NET 7
* SonarScanner for .NET version: Sonar Server Version 9.9.1 (build 69595) / SonarLint Version 7.3.0.77872
* Operating System: Windows 10

**Comment by zsolt-kolbay-sonarsource** [comment]
Thank you for reporting this issue. Confirmed as False Positive.

**Comment by martin-strecker-sonarsource** [comment]
Related #6894
Existing test case
https://github.com/SonarSource/sonar-dotnet/blob/7d09f454d9baabfed19bd11edd63891ed1aa52f0/analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ArrayPassedAsParams.cs#L98
